cmd/kubeadm/app/apis/kubeadm/v1alpha1/conversion_test.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1_test
```

Comment:
?
Why not just `v1alpha1`?

Reply:
I do this to enforce that the tests only test the public interfaces.

Reply:
Why do we have to restrict ourselves to only test the public interfaces?

Reply:
Testing the interface allows the implementation to change. If we test the implementation we are restricted to that particular implementation (unless you also want to refactor the tests at the same time).

Here is an article that sums up the argument https://testing.googleblog.com/2013/08/testing-on-toilet-test-behavior-not.html.

Reply:
To answer your question directly, we restrict ourselves so that our intention in only testing the public interfaces, unless there is good reason, is very clear.

Reply:
Okay, I think we agree here but talk past each other. However, I don't have time to argue more on this nit, this is fine by me ;)